frontend/src/scenes/funnels/funnelLogic.test.ts

```diff
@@ -183,4 +188,91 @@ describe('funnelLogic', () => {
+                    },
+                },
+                onLogic: (l) => (logic = l),
+            })
+
+            it('no query to load results', async () => {
+                await expectLogic(logic)
+                    .toMatchValues({
+                        areFiltersValid: true,
+                        results: ['cached result'],
+                        filters: expect.objectContaining({
+                            events: [{ id: 2 }, { id: 3 }],
+                            properties: [expect.objectContaining({ type: 'lol' })],
+                        }),
+                    })
+                    .toDispatchActions(['loadResultsSuccess']) // this took the cached results
+                    .toMatchValues({
+                        results: ['cached result'], // should not have changed
+                        filters: expect.objectContaining({
+                            events: [{ id: 2 }, { id: 3 }],
+                            properties: [expect.objectContaining({ value: 'lol' })],
+                        }),
+                    })
+            })
+        })
+
+        describe('props with filters, no cached results', () => {
+            initKeaTestLogic({
+                logic: funnelLogic,
+                props: {
+                    dashboardItemId: 123,
+                    cachedResults: undefined,
+                    filters: {
+                        events: [{ id: 2 }, { id: 3 }],
+                        properties: [{ value: 'a', operator: PropertyOperator.Exact, key: 'a', type: 'a' }],
+                    },
+                },
+                onLogic: (l) => (logic = l),
+            })
+
+            it('makes a query to load the results', async () => {
+                await expectLogic(logic)
+                    .toDispatchActions(['loadResultsSuccess'])
+                    .toMatchValues({
+                        results: ['result from api'],
+                        filters: expect.objectContaining({
+                            events: [{ id: 2 }, { id: 3 }],
+                            properties: [expect.objectContaining({ value: 'a' })],
+                        }),
+                    })
+            })
+
+            it('setCachedResults sets results directly', async () => {
+                await expectLogic(logic).toDispatchActions(['loadResultsSuccess'])
+
+                logic.actions.setCachedResults(
+                    {
+                        events: [{ id: 2 }, { id: 3 }],
+                        properties: [{ value: 'lol', operator: PropertyOperator.Exact, key: 'lol', type: 'lol' }],
+                    },
+                    ['cached result' as any] as FunnelAPIResponse
+                )
+
+                await expectLogic(logic)
+                    .toDispatchActions(['setCachedResults', 'setCachedResultsSuccess'])
+                    .printActions({ compact: true })
```

Comment:
I might be misunderstanding the usage of printActions, but is it more than just a debug tool? Should we remove this?

Reply:
oops, removed